camel-util/src/main/java/org/apache/camel/util/ObjectHelper.java

```diff
@@ -485,34 +485,21 @@ public static InputStream loadResourceAsStream(String name) {
     }
 
     /**
-     * Attempts to load the given resource as a stream using the thread context
-     * class loader or the class loader used to load this class
+     * Attempts to load the given resource as a stream using 
+     * first the given class loader, then the thread context
+     * class loader and finally the class loader used to load this class
      *
      * @param name the name of the resource to load
      * @param loader optional classloader to attempt first
      * @return the stream or null if it could not be loaded
      */
     public static InputStream loadResourceAsStream(String name, ClassLoader loader) {
-        InputStream in = null;
-
-        String resolvedName = resolveUriPath(name);
-        if (loader != null) {
-            in = loader.getResourceAsStream(resolvedName);
-        }
-        if (in == null) {
-            ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
-            if (contextClassLoader != null) {
-                in = contextClassLoader.getResourceAsStream(resolvedName);
-            }
-        }
-        if (in == null) {
-            in = ObjectHelper.class.getClassLoader().getResourceAsStream(resolvedName);
-        }
-        if (in == null) {
-            in = ObjectHelper.class.getResourceAsStream(resolvedName);
-        }
-
-        return in;
+    	try {
+        	URL res = loadResourceAsURL(name);
```

Comment:
I looked into the JDK source and its loadResourceAsStream does the same here to load via URL first and then open its stream. So we can safely do this also in Camel.